pkg/parser/validate/parameters.go

```diff
@@ -115,8 +115,7 @@ func (val Validate) checkParamUsedWithParam(param ast.ParameterValue, stepName s
 		// check already done before in `CheckIfParamsExist`
 		return
 	}
-
-	if paramUsedAsValue.GetType() != definedParam.GetType() {
+	if (!(definedParam.GetType() == "string" && paramUsedAsValue.GetType() == "enum")) && (paramUsedAsValue.GetType() != definedParam.GetType()) {
```

Comment:
There's a lot happening here in just one line, that makes it hard to reason about. How about adding a comment, and making the code more explicit?

```suggestion
	definedType := definedParam.GetType()
	valueType := paramUsedAsValue.GetType()
	if ((definedType != valueType)
		&& !(definedType == "string" && valueType == "enum")) { // String params can accept "string" or "enum"
```